src/backgroundblurvideoframeprocessor/BackgroundBlurVideoFrameProcessor.ts

```diff
@@ -0,0 +1,257 @@
+import BlurStrength from './BackgroundBlurStrength';
+import BackgroundFilterPaths from './BackgroundFilterPaths';
+import BackgroundFilterSpec from './BackgroundFilterSpec';
+import ModelSpecBuilder from './ModelSpecBuilder';
+
+/** @internal */
+const CREATE_DEFAULT_MODEL_SPEC = (): ModelSpec =>
+  ModelSpecBuilder.builder().withSelfieSegmentationDefaults().build();
+
+/** @internal */
+const DEFAULT_CDN = 'https://static.sdkassets.chime.aws';
+
+/** @internal */
+const DEFAULT_PATHS: BackgroundFilterPaths = {
+  worker: `${DEFAULT_CDN}/bgblur/workers/worker.js`,
+  wasm: `${DEFAULT_CDN}/bgblur/wasm/_cwt-wasm.wasm`,
+  simd: `${DEFAULT_CDN}/bgblur/wasm/_cwt-wasm-simd.wasm`,
+};
+
+/**
+ * No-op implementation of the blur processor. An instance of this class will be returned when a user attempts
+ * to create a blur processor when it is not supported.
+ */
+/** @internal */
+class NoOpBackgroundBlurProcessor
+  extends NoOpVideoFrameProcessor
+  implements BackgroundBlurProcessor {
+  /**
+   * no-op
+   */
+  setBlurStrength(): void {}
+
+  /**
+   * no-op
+   * @returns
+   */
+  async loadAssets(): Promise<void> {
+    return;
+  }
+
+  /**
+   * no-op
+   */
+  addObserver(): void {}
+
+  /**
+   * no-op
+   */
+  removeObserver(): void {}
+}
+
+/**
+ * [[BackgroundBlurProcessorFactory]]
+ * Creates a background blur processor which identifies the foreground person and blurs the background.
+ */
+export default class BackgroundBlurVideoFrameProcessor {
+  /**
+   * A factory method that will call the private constructor to instantiate the processor and asynchronously
+   * initialize the worker, wasm, and ML models. Upon completion of the initialization the promise will either
+   * be resolved or rejected.
+   * @param spec The spec defines the assets that will be used for adding background blur to a frame
+   * @param blurStrength How much blur to apply to a frame
+   * @returns
+   */
+  static async create(
+    spec?: BackgroundFilterSpec,
+    options?: BackgroundBlurOptions
+  ): Promise<BackgroundBlurProcessor | undefined> {
+    spec = BackgroundBlurVideoFrameProcessor.resolveSpec(spec);
+    options = BackgroundBlurVideoFrameProcessor.resolveOptions(options);
+    const { logger } = options;
+
+    const supported = await BackgroundBlurVideoFrameProcessor.isSupported(spec, options);
+    // if blur is not supported do not initialize. The processor will become a no op if not supported
+    logger.info(`processor is ${supported ? '' : 'not'} supported`);
+    if (!supported) {
+      logger.info('Using no-op processor because background blur is not supported');
+      return new NoOpBackgroundBlurProcessor();
+    }
+
+    let processor: BackgroundBlurProcessor;
+    if (await BackgroundBlurProcessorProvided.isSupported()) {
+      logger.info('Using browser-provided background blur');
+      processor = new BackgroundBlurProcessorProvided(spec, options);
+    } else {
+      logger.info('Using built-in background blur');
```

Comment:
nit: I think we log this in the built in processor constructor - can we just keep one of the two.